consensus/state.go

```diff
@@ -713,9 +716,10 @@ func (cs *ConsensusState) handleMsg(mi msgInfo) {
 	default:
 		cs.Logger.Error("Unknown msg type", reflect.TypeOf(msg))
 	}
-	if err != nil {
-		cs.Logger.Error("Error with msg", "height", cs.Height, "round", cs.Round, "type", reflect.TypeOf(msg), "peer", peerID, "err", err, "msg", msg)
-	}
+	// if err != nil {
+	// 	cs.Logger.Error("Error with msg", "height", cs.Height, "round", cs.Round,
```

Comment:
uncommentting it will lead to TestReactorValidatorSetChanges failing with 5m0s timeout

Reply:
@ebuchman any ideas why this may be the case?

Reply:
`TestReactorValidatorSetChanges` uses a `TestingLoggerWithOutput` which inits a `NewSyncWriter`. Could it be the case that some other go-routine is writing to stdout such that this blocks and never becomes available? 

I could not reproduce this locally so not sure what is going on.

Reply:
I removed the `-v` flag from go test, which triggerst the test to be run with a `NewNopLogger`.  Interestingly, the test passes (only tried once though): https://github.com/tendermint/tendermint/compare/develop...non-verbose/anton/ineffassign?expand=1

Reply:
Maybe it has something to do with `reflect.TypeOf`

Reply:
Can't replicate it anymore. Super weird

Reply:
`reflect.TypeOf` doesn't block on anything. The logger potentially does as there is a lock involved (I have to admit I do not understand how and why but it seems more plausible than reflect).